### PR TITLE
Use float for ordering, add Ordering module

### PR DESCRIPTION
**Pull request by glennsl**
Naively converting JS code to rescript may result in unexpected and hard to track down bugs because the `int`s used for ordering overflows much easier than JS numbers.

This does two things:
1. Uses `float`s instead of `int`s for ordering to avoid premature overflows.
2. Adds an `Ordering` module to make  comparison function code more readable, and more convenient to convert from `int`s.

### CODE REVIEW

**Review by zth — APPROVED** [review]
This looks great to me! @cknitt ?

**Comment by cknitt** [comment]
I already suggested using floats once, see #74, but @cristianoc was not in favor of that at the time.

**Comment by zth** [comment]
Right, I forgot about that conversation it seems... 🙈

File: src/Core__Ordering.res [hunk 1]
```diff
@@ -0,0 +1,13 @@
+type t = float
+
+let less = -1.
+let equal = 0.
+let greater = 1.
```
Comment (cknitt) [inline src/Core__Ordering.res:5]: I think these should have `@inline`.

Comment (glennsl) [reply src/Core__Ordering.res]: Good idea! Done in https://github.com/rescript-association/rescript-core/pull/149/commits/65028a7c41dd2d773d6f5efdba7cb0d755f9f185

**Comment by glennsl** [comment]
> I already suggested using floats once, see #74, but @cristianoc was not in favor of that at the time.

Huh, I'd  forgotten about that too. Seems I've changed my mind. Maybe @cristianoc has too!

File: src/Core__Array.res [hunk 2]
```diff
@@ -57,10 +57,10 @@ let equal = (a, b, eq) => {
 
 let rec compareFromIndex = (a, b, i, cmp, len) =>
   if i === len {
-    0
+    Core__Ordering.equal
   } else {
     let c = cmp(a->getUnsafe(i), b->getUnsafe(i))
-    if c === 0 {
+    if c->Core__Ordering.isEqual {
```
Comment (cknitt) [inline src/Core__Array.res:63]: If we do `c === Core__Ordering.equal` here instead then we save one function call and have less changes in the JS output. Same in Core_List.

**Comment by cristianoc** [comment]
> > I already suggested using floats once, see #74, but @cristianoc was not in favor of that at the time.
> 
> Huh, I'd forgotten about that too. Seems I've changed my mind. Maybe @cristianoc has too!

I just made an observation not a recommendation.
Happy to go with whatever decision now that the relevant facts are known.

File: src/Core__Array.res (hunk 2 above)
Comment (glennsl) [reply src/Core__Array.res]: Done in https://github.com/rescript-association/rescript-core/pull/149/commits/13c38da56fb32fda4dda00c0f7814ab2833b8c80

**Review by cknitt — APPROVED** [review]

**Comment by jmagaram** [comment]
Please see https://github.com/rescript-association/rescript-core/issues/150

**Pull request merged**